dask/dataframe/core.py

```diff
@@ -4694,6 +4762,7 @@ def set_index(
             pre_sorted = True
             divisions = other.divisions
 
+        # Branch according to sortedness
```

Comment:
I don't think this comment adds any understanding. Can you explain why this is happening here, or remove it?

Reply:
Changed this to 

> If index is already sorted, take advantage of that with set_sorted_index